post-install/sleep.md

```diff
@@ -0,0 +1,272 @@
+
+This guide also includes some other fixes than just mapping:
+
+* [Fixing USB Power](https://dortania.github.io/USB-Map-Guide/misc/power.md)
+* [Fixing Shutdown/Restart](https://dortania.github.io/USB-Map-Guide/misc/shutdown.md)
+* [GPRW/UPRW/LANC Instant Wake Patch](https://dortania.github.io/USB-Map-Guide/misc/instant-wake.md)
+* [Keyboard Wake Issues](https://dortania.github.io/USB-Map-Guide/misc/keyboard.md)
+
+### Fixing GPUs
+
+With GPUs, it's fairly easy to know what might be causing issues. This being unsupported GPUs in macOS. By default, any GPU that doesn't have drivers already provided in the OS will run off very basic drivers known as VESA drivers. These provide minimal display output but also cause a big issue in that macOS doesn't actually know how to properly interact with these devices. To fix this, well need to either trick macOS into thinking it's a generic PCIe device(which it can better handle, ideal for desktops) or completely power off the card(on laptops, desktop dGPUs have issues powering down)
+
+* See here for more info:
+    * [Disabling desktop dGPUs](https://dortania.github.io/Getting-Started-With-ACPI/Desktops/desktop-disable.md)
+    * [Disabling laptop dGPUs](https://dortania.github.io/Getting-Started-With-ACPI/Laptops/laptop-disable.md)
+	
+Special notes for iGPU users on 10.15.4 and newer:
+
+* iGPU wake is partially broken due to numerous hacks apple uses in AppleGraphicsPowerManagement.kext with real Macs, to get around this you'll likely need `igfxonln=1` to force all displays online. Obviously test first to make sure you have this issue.
+* AAPL,ig-platform-id `07009B3E` may fail for desktop Coffee Lake (UHD 630) users, you can try `00009B3E` instead
+
+Special note for 4k Displays with AMD dGPUs:
+
+* Some displays may fail to wake randomly, mainly caused by AGDC preferences. To fix, apply this to your dGPU in DeviceProperties:
+  * `CFG,CFG_USE_AGDC | Data | 00`
+  * You can find the PciRoot of your GPU with [gfxutil](https://github.com/acidanthera/gfxutil/releases)
+    * `/path/to/gfxutil -f GFX0`
+
+![](/images/post-install/sleep-md/agdc.png)
+
+### Fixing Thunderbolt
+
+Thunderbolt is a very tricky topic in the community, mainly due to the complexity of the situation. You really have just 2 paths to go down if you want Thunderbolt and sleep to work simultaneously:
+
+* Disable Thunderbolt 3 in the BIOS
+* Attempt to patch Thunderbolt 3:
+  * [Thunderbolt 3 Fix](https://osy.gitbook.io/hac-mini-guide/details/thunderbolt-3-fix/)
+  * [ThunderboltReset](https://github.com/osy86/ThunderboltReset)
+  * [ThunderboltPkg](https://github.com/al3xtjames/ThunderboltPkg/blob/master/Docs/FAQ.md)
+
+Note: Thunderbolt can be enabled without extra work *if* you're ok without sleep, and vice versa.
+
+### Fixing NICs
+
+NICs(network Interface Controllers) are fairly easy to fix with sleep, it's mainly the following:
+
+* Disable `WakeOnLAN` in the BIOS
+  * Most hack kexts don't support WOL
```

Comment:
Not really, but it's just that it keeps the computers in a wake-sleep loop. Especially with laptops.

Reply:
Mieze's realtek drivers support WOL technically but it's not usable because of ACPI issues.